src/bin/common/parse_command.ml

```diff
@@ -0,0 +1,1157 @@
+
+  let profiling_plugin =
+    let doc = "Use the given profiling plugin." in
+    let docv = "PGN" in
+    Arg.(value & opt string (get_profiling_plugin ()) &
+         info ["profiling-plugin"] ~docv ~docs ~doc) in
+
+  let get_verbose =
+    let doc = "Set the get_verbose mode." in
+    Arg.(value & flag & info ["v"; "verbose"] ~doc) in
+
+  Term.(ret (const mk_profiling_opt $
+             cumulative_time_profiling $ profiling $
+             profiling_plugin $ get_verbose
+            ))
+
+let parse_quantifiers_opt =
+
+  let docs = s_quantifiers in
+
+  let greedy =
+    let doc = "Use all available ground terms in instantiation." in
+    Arg.(value & flag & info ["g"; "greedy"] ~doc) in
+
+  let instantiate_after_backjump =
+    let doc =
+      "Make a (normal) instantiation round after every backjump/backtrack." in
+    Arg.(value & flag & info ["inst-after-bj"] ~docs ~doc) in
+
+  let max_multi_triggers_size =
+    let doc = "Max number of terms allowed in multi-triggers." in
+    let docv = "VAL" in
+    Arg.(value & opt int (get_max_multi_triggers_size ()) &
+         info ["max-multi-triggers-size"] ~docv ~docs ~doc) in
+
+  let nb_triggers =
+    let doc = "Number of (multi)triggers." in
+    let docv = "VAL" in
+    Arg.(value & opt int (get_nb_triggers ()) &
+         info ["nb-triggers"] ~docv ~docs ~doc) in
+
+  let no_ematching =
+    let doc = "Disable matching modulo ground equalities." in
+    Arg.(value & flag & info ["no-ematching"] ~docs ~doc) in
+
+  let no_user_triggers =
+    let doc = "Ignore user triggers, except for triggers of theories axioms"; in
+    Arg.(value & flag & info ["no-user-triggers"] ~docs ~doc) in
+
+  let normalize_instances =
+    let doc =
+      "Normalize generated substitutions by matching w.r.t. the state of \
+       the theory. This means that only terms that \
+       are greater (w.r.t. depth) than the initial terms of the problem are \
+       normalized." in
+    Arg.(value & flag & info ["normalize-instances"] ~docs ~doc) in
+
+  let triggers_var =
+    let doc = "Allows variables as triggers." in
+    Arg.(value & flag & info ["triggers-var"] ~docs ~doc) in
+
+  Term.(ret (const mk_quantifiers_opt $ greedy $ instantiate_after_backjump $
+             max_multi_triggers_size $ nb_triggers $
+             no_ematching $ no_user_triggers $ normalize_instances $
+             triggers_var
+            ))
+
+let parse_sat_opt =
+
+  let docs = s_sat in
+
+  let get_bottom_classes =
+    let doc = "Show equivalence classes at each bottom of the sat." in
+    Arg.(value & flag & info ["bottom-classes"] ~docs ~doc) in
+
+  let disable_flat_formulas_simplification =
+    let doc = "Disable facts simplifications in satML's flat formulas." in
+    Arg.(value & flag &
+         info ["disable-flat-formulas-simplification"] ~docs ~doc) in
+
+  let enable_restarts =
+    let doc =
+      "For satML: enable restarts or not. Default behavior is 'false'." in
```

Comment:
For this kind of option the widely accepted consensus for good use is to not have a flag, but instead an option taking as argument a boolean value. This helps in two ways:
- if/when the default changes, this solves a lot of headaches with introducing a `no-foobar` option
- the default value is then naturally documented since there must be default value for the boolean argument.